truffle/src/main/java/org/jruby/truffle/nodes/ext/BigDecimalNodes.java

```diff
@@ -1180,8 +1392,62 @@ public RubyBasicObject absSpecial(RubyBasicObject value) {
 
     }
 
+    @CoreMethod(names = "round", optional = 2)
+    public abstract static class RoundNode extends BigDecimalCoreMethodArrayArgumentsNode {
+
+        public RoundNode(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        @TruffleBoundary
+        private BigDecimal round(RubyBasicObject value, int digit, RoundingMode roundingMode) {
+            final BigDecimal valueBigDecimal = getBigDecimalValue(value);
+            final int end = -valueBigDecimal.scale();
+            final int start = valueBigDecimal.precision() + end;
+            final int newPrecision = start - digit;
+            return valueBigDecimal.round(new MathContext(newPrecision, roundingMode));
```

Comment:
Small style suggestion, but adding some vertical spacing (e.g., between the declarations and the return statement) would help a lot with legibility.